strawberryfields/gbs/plot.py

```diff
@@ -345,49 +346,72 @@ def points(
     return f
 
 
-LIGHT_BLUE = "#94cecf"
-
-BAR_COLOUR = LIGHT_BLUE
-
-
-def spectrum(energies: list, bins: Optional[int] = 0):  # pragma: no cover
-    """Plots a spectrum based on input sampled energies.
+def spectrum(
+    energies: list, gamma: float = 100.0, xmin: float = None, xmax: float = None
+):  # pragma: no cover
+    """Plots a vibronic spectrum based on input sampled energies.
 
     Args:
         energies (list[float]): a list of sampled energies
-        bins (int): maximum number of bins. If zero, the number of bins is decided automatically
-            by Plotly.
+        gamma (float): parameter specifying the width of the Lorentzian function
+        xmin (float): minimum limit of the x axis
+        xmax (float): maximum limit of the x axis
 
     Returns:
-         Figure: spectrum in the form of a histogram of energies
+         Figure: spectrum in the form of a histogram of energies with a Lorentzian-function curve
     """
     try:
         import plotly.graph_objects as go
     except ImportError:
         raise ImportError(plotly_error)
 
-    h = go.Histogram(x=energies, nbinsx=bins, marker={"color": BAR_COLOUR})
+    emin = min(energies)
+    emax = max(energies)
+    if xmin is None:
+        xmin = emin - 0.1 * (emax - emin)
+    if xmax is None:
+        xmax = emax + 0.1 * (emax - emin)
+    bins = int(emax - emin) // 5
+    bar_width = (emax - emin) * 0.005
+    line_width = 3.0
+
+    h = np.histogram(energies, bins)
+    X = np.linspace(xmin, xmax, int(xmax-xmin))
+    L = 0
+    for e in energies:
+        L += (gamma / 2)**2 / ((X - e)**2 + (gamma / 2)**2)
+
+    text_font = dict(color="black", family="Computer Modern")
```

Comment:
I'm curious what motivated the decision to move away from Plotly's `Histogram`?

Reply:
We have to have exactly same bins and counts to make sure that the bars and Lorentzian curves  are generated based on the same data. Note that we need the numeric data to build the Lorentzian smooth curve, which I could not get the numeric bin and count values from plotly.

Reply:
Ahh that makes sense! Then I'm happy with this choice! Thanks